mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -242,6 +242,28 @@ struct BlockwiseGemmRewritePattern
       return splitTidForLDS;
     };
 
+    int64_t aCopyPerThread = (k * m) / blockSize;
+    int64_t bCopyPerThread = (k * n) / blockSize;
+
+    Type elementTypeA = blockAType.getElementType();
+    if (auto vectorDataType = elementTypeA.dyn_cast<VectorType>())
+      elementTypeA = vectorDataType.getElementType();
+
+    auto maybeCopyAPerThread = computeCopyPerThread(
+        elementTypeA, aCopyPerThread, k * kPack, m, kPack, loc);
+    if (failed(maybeCopyAPerThread))
+      return maybeCopyAPerThread;
+    int64_t copyMPerThread = (*maybeCopyAPerThread).second;
```

Comment:
This feels fragile to me...
We shouldn;t be recomputing this as it was decided before the op is created and this coule be problematic if the op is created with a seperate logic.

I;d say we just put an attribute on the op to indicate mPerThread and nPerThread; I think that is all you want to know rather than recomputing it here. It makes the debugging easier in future also less problematic for the second gemm in the attention kernel.

Reply:
It is super fragile, but I feared that I was adding too many attributes on the op. But if it is fine with you, I will go ahead and do it :)